src/main/java/com/iexec/worker/pubsub/SubscriptionService.java

```diff
@@ -84,16 +86,25 @@ public void subscribeToTopic(String chainTaskId) {
     /**
      * Unsubscribe from topic if already subscribed.
      *
-     * @param chainTaskId
+     * @param chainTaskId id of the task to unsubscribe and purge
+     * @return true if unsubscribed and purge topic successfully, false otherwise
      */
-    public void unsubscribeFromTopic(String chainTaskId) {
+    @Override
+    public boolean purgeTask(String chainTaskId) {
         if (!isSubscribedToTopic(chainTaskId)) {
             log.error("Already unsubscribed from topic [chainTaskId:{}]", chainTaskId);
-            return;
+            return true;
         }
         this.chainTaskIdToSubscription.get(chainTaskId).unsubscribe();
         this.chainTaskIdToSubscription.remove(chainTaskId);
         log.info("Unsubscribed from topic [chainTaskId:{}]", chainTaskId);
+        return !isSubscribedToTopic(chainTaskId);
+    }
+
+    @Override
+    public void purgeAllTasksData() {
+        final List<String> tasksIds = new ArrayList<>(chainTaskIdToSubscription.keySet());
+        tasksIds.forEach(this::purgeTask);
```

Comment:
Here, the affectation to a variable can be avoided with a pure functional syntax.
```suggestion
        chainTaskIdToSubscription.keySet().forEach(this::purgeTask);
        chainTaskIdToSubscription.clear();
```

Reply:
Resolve in [36484b8](https://github.com/iExecBlockchainComputing/iexec-worker/pull/620/commits/36484b81174020d317b49d1b9ac290f957a192f9)